src/fdc/DiskImageUtils.cc

```diff
@@ -77,21 +77,25 @@ struct SetBootSectorResult {
 	uint8_t descriptor;
 };
 static SetBootSectorResult setBootSector(
-	MSXBootSector& boot, size_t nbSectors, bool dos1)
+	MSXBootSector& boot, MSXBootSectorType bootType, size_t nbSectors)
 {
 	// start from the default boot block ..
-	const auto& defaultBootBlock = dos1 ? BootBlocks::dos1BootBlock : BootBlocks::dos2BootBlock;
-	boot = defaultBootBlock.bootSector;
+	if (bootType == MSXBootSectorType::DOS1) {
+		boot = BootBlocks::dos1BootBlock.bootSector;
+	} else if (bootType == MSXBootSectorType::DOS2) {
+		boot = BootBlocks::dos2BootBlock.bootSector;
+	}
```

Comment:
Detail: this code makes me a bit nervous. What if `bootType` is neither of these two values?
I know that in this commit that isn't possible. But it's not robust against future changes (I haven't yet looked in later patches in this series how you adapt this code when the enum does get extended).

I'd suggest one of the following alternatives:
```
if (bootType == ...) {
} else {
    assert(bootType = ...)
}
```
```
if (bootType == ...) {
} else if (bootType = ...)
} else {
    UNREACHABLE;
}
```
```
switch (bootType) {
case ...
case ...
// NO default branch. And then rely on compiler warnings for unhandled enum values
```
My personal preference is the last alternative (if that's warning free in the 3 major compilers, I haven't checked this).